rakelib/github_stats.rake

```diff
@@ -31,15 +42,29 @@ namespace :github_stats do
       open_prs = JSON.parse(json_response)
       open_prs.each do |pr|
         # parse vals from json
+        pr_url = pr['url']
         user = pr['user']['login']
         number = pr['number']
-        repo_name = repo
-        updated_at = pr['updated_at']
+        pr_created_at = DateTime.parse(pr['created_at']).to_f
+        comments = pr['comments'].to_i
+        comments_url = pr['comments_url']
+        review_comments = pr['review_comments'].to_i
+        review_comments_url = pr['review_comments_url']
+        reviews_url = "#{pr_url}/reviews"
+        # determine first_responded_at date
+        first_responded_at = if comments.positive?
+                               get_first_responded_at(comments_url)
+                             elsif review_comments.positive?
+                               get_first_responded_at(review_comments_url)
+                             else
+                               get_first_responded_at(reviews_url)
+                             end
```

Comment:
What about a scenario where a review occurs, then sometime later a regular comment occurs? This if-else logic block would determine that the comment came first even though the review came before it. Am I reading that right?

Reply:
Is it correct to assume that if a comment exists, that it must have come first?

Reply:
Yeah, I think you're right. In the first hit that pulls everything, we have access to comment counts and review_comment counts for each PR but, for whatever reason, not review counts. In trying to limit the number of times we have to hit the API I figured I'd check the comment and review_comment counts first, since if one of them is positive then we hit the specific endpoint for it. If both are zero then we can hit the review endpoint and determine if there are either zero reviews (nothing has been done at all with the PR yet) or one or more reviews, in which case we grab the first and get its date. We'll have to increase the number of hits to the API but I think I know what I need to do to account for the scenario you describe.

Reply:
After discussing with my group, we determined that we should only calculate duration based on actual reviews and not on comments or review_comments, which fortunately makes this a lot simpler.

Reply:
Interesting. I can provide my own anecdote that I sometimes will first respond with plain old comments if I don't understand something. I find that getting all my comments in, and then submitting my review might take as long as 45 minutes! Whereas if I make simple comments one by one, the author may respond in real time as I am still reviewing.